agent/wptdriver/wpt_driver_core.cc

```diff
@@ -972,4 +974,21 @@ bool WptDriverCore::NeedsReboot() {
   }
 
   return _exit;
-}
\ No newline at end of file
+}
+
+bool WptDriverCore::NeedsRestart()

```

Comment:
Would prefer if you called the method something like RestartIfNeeded as the method actually does the restart